File: cpp/sorting/wiggle_sort.cpp
#include<iostream>
#include<vector>
#include<algorithm>
using namespace std; 

void display(int*array, int size) {
for(int value=0; value<size; value++)
cout << array[value] << " ";
cout << endl;
}
void swap(int *x, int *y) { 
    int temp = *x; 
    *x = *y; 
    *y = temp; 
} 
// Wiggle Sort -> {arr[0] >= arr[1] <= arr[2] >= arr[3] <= arr[4] >= arr[5]}
void WiggleSort(int arr[], int n) 
{ 
    // Sort the input array 
    sort(arr, arr+n); 
    for (int i=0; i<n-1; i += 2) 
        swap(&arr[i], &arr[i+1]); 
} 
int main() {
  int n;
  cout << "Enter the number of elements: ";
  cin >> n;
  int arr[n];
  cout << "Enter the elements value:" << endl;
  for (int i = 0; i < n; i++) {
    cin >> arr[i];
  }
  cout << "Array before Wiggle Sorting: ";
  display(arr, n);
  WiggleSort(arr, n);
  cout << "Array after Wiggle Sorting: ";
  display(arr, n);
}

File: java/sorting/wiggle_sort.java

import java.util.*;

class sortInwiggle
{

    void swap(int arr[], int a, int b)
    {
        int temp = arr[a];
        arr[a] = arr[b];
        arr[b] = temp;
    }

    void sortInwiggle(int arr[], int n)
    {
        Arrays.sort(arr);
        for (int i=0; i<n-1; i += 2)
            swap(arr, i, i+1);
    }
    
    public static void main(String args[])
    {
        SortWave ob = new sortInwiggle();
        int arr[] = {10, 90, 49, 2, 1, 5, 23};
        int n = arr.length;
        ob.sortInwiggle(arr, n);
        for (int i : arr)
            System.out.print(i + " ");
    }
}
